validation/src/test/scala/hmda/validation/rules/lar/quality/Q027Spec.scala

```diff
@@ -0,0 +1,42 @@
+package hmda.validation.rules.lar.quality
+
+import hmda.model.fi.lar.LoanApplicationRegister
+import hmda.validation.rules.EditCheck
+import hmda.validation.rules.lar.LarEditCheckSpec
+
+class Q027Spec extends LarEditCheckSpec {
+  property("fails when actionTaken = 1,2,3,4,5,7,8 and propertyType = 1,2 and income NA") {
+    forAll(larGen) { lar =>
+      whenever(lar.actionTakenType != 6 && lar.loan.propertyType != 3) {
+        val invalidApplicant = lar.applicant.copy(income = "NA")
+        val invalidLar = lar.copy(applicant = invalidApplicant)
+        invalidLar.mustFail
+      }
+    }
+  }
+
+  property("passes when applicant income not NA") {
+    forAll(larGen) { lar =>
+      whenever(lar.applicant.income != "NA") {
+        lar.mustPass
+      }
+    }
+  }
+
+  property("passes when action taken not 1,2,3,4,5,7,8") {
+    forAll(larGen) { lar =>
+      val validLar = lar.copy(actionTakenType = 6)
```

Comment:
@schbetsy I think that is an acceptable way to test this edit. For those that have values not contained in a field (i.e. a fixed $ amount) we decided to put them in configuration, as they do change through time. As @debseidner said, if the file spec changes and we accept new or different values, our codebase would need to be updated.